crates/rattler_repodata_gateway/src/gateway/query.rs

```diff
@@ -120,8 +122,16 @@ impl GatewayQuery {
         // Package names that we have or will issue requests for.
         let mut seen = HashSet::new();
         let mut pending_package_specs = HashMap::new();
+        let mut direct_url_specs = vec![];
         for spec in self.specs {
-            if let Some(name) = &spec.name {
+            if spec.url.is_some() {
+                let name = spec
+                    .name
+                    .clone()
+                    .expect("direct url spec should always have a name");
```

Comment:
Im thinking if we should turn this into an error that is returned instead of expecting. You can just create a matchspec manually that doesnt have this property.